src/ipc/ipc4/helper.c

```diff
@@ -91,7 +95,15 @@ struct comp_dev *comp_new(struct sof_ipc_comp *comp)
 	dcache_invalidate_region((__sparse_force void __sparse_cache *)(MAILBOX_HOSTBOX_BASE),
 				 MAILBOX_HOSTBOX_SIZE);
 
-	dev = drv->ops.create(drv, &ipc_config, (void *)MAILBOX_HOSTBOX_BASE);
+	if (drv->type == SOF_COMP_DYNAMIC) {
+		//spec.config = (void *)MAILBOX_HOSTBOX_BASE;
+		spec.data = (unsigned char *)MAILBOX_HOSTBOX_BASE;
+		/* spec_size in IPC4 is in DW. Convert to bytes. */
+		spec.size = (comp->ext_data_length * 4);
+		dev = drv->ops.create(drv, &ipc_config, (void *)&spec);
```

Comment:
don't think the typecast is needed

Reply:
Is needed overwise there are compilation errors.